kitsune/search/views.py

```diff
@@ -119,176 +113,41 @@ def simple_search(request, template=None):
-                .indexes(es_utils.read_index('default')))
-
-    wiki_f = F(model='wiki_document')
-    question_f = F(model='questions_question')
-
-    cleaned_q = cleaned['q']
-    products = cleaned['product']
-
-    if not products and 'all_products' not in request.GET:
-        lowered_q = cleaned_q.lower()
-
-        if 'thunderbird' in lowered_q:
-            products.append('thunderbird')
-        elif 'android' in lowered_q:
-            products.append('mobile')
-        elif ('ios' in lowered_q or 'ipad' in lowered_q or 'ipod' in lowered_q or
-              'iphone' in lowered_q):
-            products.append('ios')
-        elif 'firefox os' in lowered_q:
-            products.append('firefox-os')
-        elif 'firefox' in lowered_q:
-            products.append('firefox')
-
-    # Start - wiki filters
-
-    if cleaned['w'] & constants.WHERE_WIKI:
-        # Category filter
-        wiki_f &= F(document_category__in=settings.SEARCH_DEFAULT_CATEGORIES)
-
-        # Locale filter
-        wiki_f &= F(document_locale=language)
-
-        # Product filter
-        for p in products:
-            wiki_f &= F(product=p)
-
-        # Archived bit
-        wiki_f &= F(document_is_archived=False)
-
-    # End - wiki filters
-
-    # Start - support questions filters
-
-    if cleaned['w'] & constants.WHERE_SUPPORT:
-        # Has helpful answers is set by default if using basic search
-        cleaned['has_helpful'] = constants.TERNARY_YES
-
-        # No archived questions in default search.
-        cleaned['is_archived'] = constants.TERNARY_NO
-
-        # These filters are ternary, they can be either YES, NO, or OFF
-        ternary_filters = ('has_helpful', 'is_archived')
-        d = dict(('question_%s' % filter_name,
-                  _ternary_filter(cleaned[filter_name]))
-                 for filter_name in ternary_filters if cleaned[filter_name])
-        if d:
-            question_f &= F(**d)
-
-        # Product filter
-        for p in products:
-            question_f &= F(product=p)
-
-    # End - support questions filters
-
-    # Done with all the filtery stuff--time  to generate results
-
-    # Combine all the filters and add to the searcher
-    doctypes = []
-    final_filter = F()
-    if cleaned['w'] & constants.WHERE_WIKI:
-        doctypes.append(DocumentMappingType.get_mapping_type_name())
-        final_filter |= wiki_f
-
-    if cleaned['w'] & constants.WHERE_SUPPORT:
-        doctypes.append(QuestionMappingType.get_mapping_type_name())
-        final_filter |= question_f
-
-    searcher = searcher.doctypes(*doctypes)
-    searcher = searcher.filter(final_filter)
-
-    if 'explain' in request.GET and request.GET['explain'] == '1':
-        searcher = searcher.explain()
+    lang_name = settings.LANGUAGES_DICT.get(lang) or ''
 
-    documents = ComposedList()
+    searcher = generate_simple_search(search_form, language, with_highlights=True)
+    searcher = searcher[:settings.SEARCH_MAX_RESULTS]
```

Comment:
I had no idea we did this. I always thought it was curious how the search pages maxed out at 1000. Turns out *we* did that.
![https://media2.giphy.com/media/Y2nbrJyAR6RiM/200_s.gif](https://media2.giphy.com/media/Y2nbrJyAR6RiM/200_s.gif)